AppTaskQueue/appscale/taskqueue/queue.py

```diff
@@ -675,6 +644,109 @@ def to_json(self, include_stats=False, fields=None):
 
     return json.dumps(queue)
 
+  def _task_matches_params(self, parameters):
+    """ Checks if the task entry matches the given parameters.
+
+    Args:
+      parameters: A dictionary specifying the task parameters.
+    Returns:
+      A boolean indicating whether or not the task matches the parameters.
+    """
+    parameters.update({'app': self.app, 'queue': self.name})
+    select_statement = SimpleStatement("""
+      SELECT * FROM pull_queue_tasks
+      WHERE app = %(app)s AND queue = %(queue)s AND id = %(id)s
+    """, consistency_level=ConsistencyLevel.SERIAL)
+    try:
+      result = self.db_access.session.execute(select_statement, parameters)[0]
+    except IndexError:
+      return False
+
+    for parameter in parameters:
+      if parameters[parameter] != getattr(result, parameter):
+        return False
+
+    return True
+
+  def _insert_task(self, parameters, retries=5):
+    """ Insert task entry into pull_queue_tasks.
+
+    Args:
+      parameters: A dictionary specifying the task parameters.
+      retries: The number of times to try the insert.
+    Raises:
+      InvalidTaskInfo if the task ID already exists in the queue.
+    """
+    insert_statement = SimpleStatement("""
+      INSERT INTO pull_queue_tasks (
+        app, queue, id, payload,
+        enqueued, lease_expires, retry_count, tag
+      )
+      VALUES (
+        %(app)s, %(queue)s, %(id)s, %(payload)s,
+        %(enqueued)s, %(lease_expires)s, %(retry_count)s, %(tag)s
+      )
+      IF NOT EXISTS
+    """, retry_policy=NO_RETRIES)
+    try:
+      result = self.db_access.session.execute(insert_statement, parameters)
+    except TRANSIENT_CASSANDRA_ERRORS as error:
+      retries_left = retries - 1
+      if retries_left <= 0:
+        raise
+      logger.warning(
+        'Encountered error while inserting task: {}. Retrying.'.format(error))
+      return self._insert_task(parameters, retries=retries_left)
+
+    if result.was_applied:
+      return
+
+    if not self._task_matches_params(parameters):
+      raise InvalidTaskInfo(
+        'Task name already taken: {}'.format(parameters['id']))
+
+  def _update_lease(self, parameters, check_lease=True, retries=5):
+    """ Update lease expiration on a task entry.
+
+    Args:
+      parameters: A dictionary specifying the new parameters.
+      check_lease: A boolean specifying that the old lease_expires field must
+        match the one provided.
+      retries: The number of times to try the update.
```

Comment:
@cdonati , it makes sense to extend `update_lease` and `update_task` methods with this parameter. And, as it should be optional for those, I'd suggest to make it required on the level of private method, and specify default values on the level of API methods. It's safer to have a single default instead of multiple across different methods. The same for insert task.